src/dom_types.rs

```diff
@@ -653,17 +662,80 @@ impl<Ms: 'static, OtherMs: 'static> MessageMapper<Ms, OtherMs> for Vec<El<Ms>> {
 }
 
 impl<Ms> ElContainer<Ms> for El<Ms> {
-    fn els(self) -> Vec<El<Ms>> {
-        vec![self]
+    fn els(self) -> Vec<Node<Ms>> {
+        vec![Node::Element(self)]
     }
 }
 
 impl<Ms> ElContainer<Ms> for Vec<El<Ms>> {
-    fn els(self) -> Vec<El<Ms>> {
+    fn els(self) -> Vec<Node<Ms>> {
+        self.into_iter().map(Node::Element).collect()
+    }
+}
+
+impl<Ms: 'static> ElContainer<Ms> for Node<Ms> {
+    fn els(self) -> Vec<Node<Ms>> {
+        vec![self]
+    }
+}
+
+impl<Ms: 'static> ElContainer<Ms> for Vec<Node<Ms>> {
+    fn els(self) -> Vec<Node<Ms>> {
         self
     }
 }
 
+/// For representing text nodes.
+#[derive(Clone, Debug)]
+pub struct Text {
+    pub text: String,
+    pub node_ws: Option<web_sys::Node>,
+}
+
+impl PartialEq for Text {
+    fn eq(&self, other: &Self) -> bool {
+        self.text == other.text
+    }
+}
+
+impl Text {
+    pub fn new(text: String) -> Self {
+        Self {
+            text,
+            node_ws: None,
+        }
+    }
+}
+
+/// An component in our virtual DOM. Related to, but different from
+/// [DOM Nodes](https://developer.mozilla.org/en-US/docs/Web/API/Node/nodeType)
+#[derive(Clone, Debug, PartialEq)]
+pub enum Node<Ms: 'static> {
+    Element(El<Ms>),
+    //    Svg(El<Ms>),  // May be best to handle using namespace field on El
```

Comment:
As far as I can tell, SVG elements share most of the fields of normal ones, the main diff being the namespace. The main appeal to me of this refactor was `Empty` and `Text` previously including a bunch of spurious data that we told the program to ignore. What do you think? I'm not experienced with SVG.

Reply:
Ok, let's leave it at that and refactor it when necessary. I was just afraid that something like this:
```rust
fn is_svg(el) -> bool {
    el.namespace.is_some();
}
```
will appear in the code.

Reply:
An alternative is to add the SVG Node variant, and have it wrap an Element. We'd then use an El method for namespace instead of a field.

Reply:
Could you write a short code example? I'm not 100% sure if I understand correctly.

Reply:
```rust
pub enum Node<Ms: 'static> {
    Element(El<Ms>),
    Svg(El<Ms>)
```
Leaning against it though, since it'd add more repetition to the patch logic, since as far as I know, the patching/tree logic works the same for SVG as Elements.

Reply:
Ok, we can change it in the future - it seems that there are more disadvantages than benefits at the moment and this PR is complex enough even without this.

Reply:
**RESOLVED**